python/mxnet/test_utils.py

```diff
@@ -79,10 +79,117 @@ def random_sample(population, k):
     return population_copy[0:k]
 
 
-def rand_sparse_ndarray(shape, stype, density=None):
-    """Generate a random sparse ndarray. Returns the ndarray, value(np) and indices(np) """
+def _get_uniform_dataset_csr(num_rows, num_cols, density=0.1):
+    """Returns CSRNDArray with uniform distribution
+    """
+    if num_rows <= 0 or num_cols <= 0:
+        raise ValueError("num_rows or num_cols should be greater than 0")
+
+    if density < 0 or density > 1:
+        raise ValueError("density has to be between 0 and 1")
+
+    return mx.nd.array(sp.rand(num_rows, num_cols, density).toarray())._to_csr()
+
+
+def _get_powerlaw_dataset_csr(num_rows, num_cols, density=0.1):
+    """Returns CSRNDArray with powerlaw distribution
+    with exponentially increasing number of non zeros in each row.
+    Not supported for cases where totalnnz < 2*num_rows. This is because
+    the algorithm first tries to ensure that there are rows with no zeros by
+    putting non zeros at beginning of each row.
+    """
+
+    def validate_inputs(totalnnz, num_rows, num_cols):
+        """Validate inputs"""
+        if num_rows <= 0 or num_cols <= 0:
+            raise ValueError("num_rows or num_cols should be greater than 0")
+
+        if density < 0 or density > 1:
+            raise ValueError("density has to be between 0 and 1")
```

Comment:
since those are the same checks for rows, cols, and density as the ones for uniform, would it make sense to validate in a separate function used by both uniform and powerlaw?